packages/changed-elements-react/src/widgets/comparisonJobWidget/common/versionCompareV2WidgetUtils.ts

```diff
@@ -0,0 +1,100 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { IModelConnection } from "@itwin/core-frontend";
+import { ComparisonJobCompleted, ComparisonJobStarted, IComparisonJobClient } from "../../../clients/IComparisonJobClient";
+import { NamedVersion } from "../../../clients/iModelsClient";
+import { VersionCompare } from "../../../api/VersionCompare";
+import { toastComparisonVisualizationStarting } from "./versionComapreToasts";
+import { Logger } from "@itwin/core-bentley";
+import { JobStatusAndJobProgress } from "../models/ComparisonJobModels";
+import { VersionState } from "../models/VersionState";
+
+export type ManagerStartComparisonV2Args = {
+  comparisonJob: ComparisonJobCompleted;
+  comparisonJobClient: IComparisonJobClient;
+  iModelConnection: IModelConnection;
+  targetVersion: NamedVersion;
+  currentVersion: NamedVersion;
+};
+
+export const runManagerStartComparisonV2 = async (args: ManagerStartComparisonV2Args) => {
+  if (VersionCompare.manager?.isComparing) {
+    return;
+  }
+  toastComparisonVisualizationStarting();
+  const changedElements = await args.comparisonJobClient.getComparisonJobResult(args.comparisonJob);
+  VersionCompare.manager?.startComparisonV2(args.iModelConnection, args.currentVersion, args.targetVersion, [changedElements.changedElements]).catch((e) => {
+    Logger.logError(VersionCompare.logCategory, "Could not start version comparison: " + e);
+  });
+};
+
+export const getJobStatusAndJobProgress = async (comparisonJobClient: IComparisonJobClient, entry: VersionState, iTwinId: string, iModelId: string, currentChangesetId: string): Promise<JobStatusAndJobProgress> => {
```

Comment:
Could organize the function parameters into multiple lines since it is getting quite long